app/vmctl/README.md

```diff
@@ -87,6 +88,27 @@ ARM build may run on Raspberry Pi or on [energy-efficient ARM servers](https://b
 2. Run `make vmctl-arm-prod` or `make vmctl-arm64-prod` from the root folder of [the repository](https://github.com/VictoriaMetrics/VictoriaMetrics).
    It builds `vmctl-arm-prod` or `vmctl-arm64-prod` binary respectively and puts it into the `bin` folder.
 
+## Migrating data from OpenTSDB
+
+*OpenTSDB migration is not possible without a functioning [meta](http://opentsdb.net/docs/build/html/user_guide/metadata.html) table to search for metrics/series.*
+
+OpenTSDB migration works like so:
+
+1. Find metrics based on selected filters (or the default filter set ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'])
+  * e.g. `curl -Ss "http://opentsdb:4242/api/suggest?type=metrics&q=sys"`
+2. Find series associated with each returned metric
+  * e.g. `curl -Ss "http://opentsdb:4242/api/search/lookup?m=system.load5&limit=1000000"`
+3. Download data for each series in chunks defined in the CLI switches
+  * e.g. `-retention=sum-1m-avg:1h:90d` ==
+    * `curl -Ss "http://opentsdb:4242/api/query?start=1h-ago&end=now&m=sum:1m-avg-none:system.load5\{host=host1\}"`
+    * `curl -Ss "http://opentsdb:4242/api/query?start=2h-ago&end=1h-ago&m=sum:1m-avg-none:system.load5\{host=host1\}"`
+    * `curl -Ss "http://opentsdb:4242/api/query?start=3h-ago&end=2h-ago&m=sum:1m-avg-none:system.load5\{host=host1\}"`
+    * ...
+    * `curl -Ss "http://opentsdb:4242/api/query?start=2160h-ago&end=2159h-ago&m=sum:1m-avg-none:system.load5\{host=host1\}"`
+
+This means that we must stream data from OpenTSDB to VictoriaMetrics in chunks. This is where concurrency for OpenTSDB comes in. We can query multiple chunks at once, but we shouldn't perform too many chunks at a time to avoid overloading the OpenTSDB cluster.
+
+OpenTSDB queries are likely to get stuck or otherwise fail. We plan to implement a "running cache" of the data as it is migrated so you can interrupt the ingest process and not have to start over from the beginning.
```

Comment:
we can probably add retries... For example, vm importer does not give up too quickly :-)
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmctl/vm/vm.go#L232

Reply:
The challenge isn't on the client side, it's within the OpenTSDB -> HBase connection where the queries get "stuck". So retries can actually exacerbate the problem. What I'm hoping is this:
1. the provided "starting timestamp" can be grabbed by the user in the case they need to stop the ingest.
2. They can then re-supply that as "hard timestamp" and restart with a reduced set of metrics to request.
It's manual, but it does allow for starting and stopping the ingester as needed.

Reply:
Oh, I see. Thanks for explanation!
VMimporter already supports the similar thing - it reports min/max timestamps of the latest block it failed to ingest. See `wrapErr` function which shows the exact timestamps. Maybe for opentsdb we can do something similar - cache the latest received timestamp and if GET request timeouts - print it to user.